### PR TITLE
feat: cancel session

### PR DESCRIPTION
**Pull request by 2fd**
#151

Screen Shot 2022-04-01 at 16 45 03

## How to test it

- Login with your wallet (Metamask or WalletConnect)
- When the application ask to sign the login message just reload de page (don't complete the sign and don't cancel it)
- Once the page is reloaded will automatically show you the new cancel option

### CODE REVIEW

**Comment by RominaMarchetti** [comment]
The UI implementation is impecable.
The only thing I noticed is that the cancel button doesn't work immediately, I have to click it twice to cancel the request.

File: src/components/auth/LoginContainer.tsx [hunk 1]
```diff
@@ -35,6 +36,11 @@ const mapDispatchToProps = (dispatch: any) => ({
   onLogin: (providerType: ProviderType | null) => {
     track('click_login_button', { provider_type: providerType || 'guest' })
     authenticate(providerType)
+  },
+  onCancelLogin: () => {
+    track('click_login_button')
```
Comment (cazala) [inline src/components/auth/LoginContainer.tsx:41]: is it okay that this is tracking the same event as the login? shouldn't be a "cancel login" event instead?

Comment (2fd) [reply src/components/auth/LoginContainer.tsx:41]: should be different

**Comment by RominaMarchetti** [comment]
@2fd I can barely see the Metamask loading with the cancel button state. It only appears for a second and it's quickly replaced for the single loading over the card. Is that okay?

**Comment by 2fd** [comment]
@RominaMarchetti did you follow the instrucción? the cancel button is active while waiting for the user to sign a message, if the user signs it instantly the button will disappear just as quickly

**Comment by RominaMarchetti** [comment]
@2fd I forgot to follow the instructions, sorry. It worked!

**Comment by cazala** [comment]
it works fine for me too! ✅

**Review by cazala — APPROVED** [review]

**Pull request merged**